src/flask/_compat.py

```diff
@@ -0,0 +1,19 @@
+import asyncio
+import functools
+import inspect
+import sys
+
+
+def iscoroutinefunction(func):
```

Comment:
I don't want to bring back the `_compat` module, I killed that when dropping Python 2. :-)

Instead, I would this after the imports in `app.py`:

```python
import inspect

if sys.version_info >= (3, 8):
    iscoroutinefunction = inspect.iscoroutinefunction
else:
    def iscoroutinefunction(func):
        while inspect.ismethod(func):
            func = func.__func__

        while isinstance(func, functools.partial):
            func = func.func

        return inspect.iscoroutinefunction(func)
```